scene/3d/voxel_gi.cpp

```diff
@@ -432,22 +442,27 @@ void VoxelGI::bake(Node *p_from_node, bool p_create_visual_debug) {
 	_find_meshes(p_from_node, mesh_list);
 
 	if (bake_begin_function) {
-		bake_begin_function(mesh_list.size() + 1);
+		bake_begin_function();
 	}
 
-	int pmc = 0;
+	Voxelizer::BakeStepFunc voxelizer_step_func = bake_step_function != nullptr ? voxelizer_plot_bake_step_function : nullptr;
 
+	voxelizer_plot_bake_total = voxelizer_plot_bake_base = 0;
 	for (PlotMesh &E : mesh_list) {
-		if (bake_step_function) {
-			bake_step_function(pmc, RTR("Plotting Meshes") + " " + itos(pmc) + "/" + itos(mesh_list.size()));
+		voxelizer_plot_bake_total += baker.get_bake_steps(E.mesh);
+	}
+	for (PlotMesh &E : mesh_list) {
+		if (baker.plot_mesh(E.local_xform, E.mesh, E.instance_materials, E.override_material, voxelizer_step_func) != Voxelizer::BAKE_RESULT_OK) {
+			baker.end_bake();
+			if (bake_end_function) {
+				bake_end_function();
+			}
+			return;
 		}
-
-		pmc++;
-
-		baker.plot_mesh(E.local_xform, E.mesh, E.instance_materials, E.override_material);
+		voxelizer_plot_bake_base += baker.get_bake_steps(E.mesh);
 	}
 	if (bake_step_function) {
-		bake_step_function(pmc++, RTR("Finishing Plot"));
+		bake_step_function(500, RTR("Finishing Plot"));
```

Comment:
The 500 here seems arbitrary. Is there a reason for this value? Keep in mind -1 can be used to allow the  step function to auto-increment the value displayed

Reply:
Think of it as percentage (500=50%), 0..500 is plotting meshes, 501...1000 is calculating SDF. "Finishing Plot" is reported when plotting is finished - i.e. half of the work is done.

That plotting and SDF calculation are of equal weight is arbitrary though it seemed to be more or less fine from the few runs i did with both devbuild and production builds.

Reply:
Okay, that makes sense, I can see you use that in `voxelizer_plot_bake_step_function()` and `voxelizer_sdf_bake_step_function()`